crates/matrix-sdk/src/widget/machine/mod.rs

```diff
@@ -500,14 +525,14 @@ impl WidgetMachine {
 
     fn send_from_widget_error_response(
         raw_request: Raw<FromWidgetRequest>,
-        error: impl fmt::Display,
+        error: FromWidgetErrorResponse,
     ) -> Action {
-        Self::send_from_widget_response(raw_request, FromWidgetErrorResponse::new(error))
+        Self::send_from_widget_response(raw_request, error)
```

Comment:
Maybe `send_from_widget_error_response` isn't required anymore, if it only forwards to another function?

Reply:
all the helper methods actually can be removed. `send_from_widget_response` is the only thing we need.